core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -103,6 +104,9 @@ object SparkSubmit {
    /___/ .__/\_,_/_/ /_/\_\   version %s
       /_/
                         """.format(SPARK_VERSION))
+    printStream.println("Branch %s".format(SPARK_BRANCH))
+    printStream.println("Compiled by user %s on %s".format(SPARK_BUILD_USER, SPARK_BUILD_DATE))
+    printStream.println("Url %s".format(SPARK_REPO_URL))
```

Comment:
I had checked on this earlier. Currently the information is displayed at multiple sources - SparkSubmit, REPLs for 2.10 and 2.11, Python and R shell's. 

I personally would prefer to keep this consistent. However I see the welcome message being duplicated at 3 places in the scala code itself with some minor differences in the messages shown. 
Refactoring them into a single location is something that we should do in the ideal case. 

Since the focus was to correctly identify the spark versions running on "real" clusters, we limited the change to SparkSubmit as users aren't allowed to fire up spark shells on the cluster. 

@jodersky, @vanzin let me know if you guys want it to have it consistently across all the messages. Or we could just accept this one now and file another JIRA for the refactoring.